### ISSUE TITLE
Provide a link to pre-select a cube dataset.

### ISSUE THREAD

**Issue by l00mi** [visualize-admin/visualization-tool#87]
To be able to have a direct link to a published cube inside visualze.admin.ch from:
* https://github.com/zazuko/cube-creator
* http://lindas.admin.ch/dataset (see also https://github.com/zazuko/cube-creator/issues/730)
* and also potentially opendata.swiss

It would be necessary to be able to compose a URI based on the Cube Dataset Uri. Ideally in a non-versioned and versioned fashion.

As illustrative proposition:
`https://visualize.admin.ch/en/create/new/https%3A%2F%2Fenvironment.ld.admin.ch%2Ffoen%2Fudb066`
and
'`https://visualize.admin.ch/en/create/new/https%3A%2F%2Fenvironment.ld.admin.ch%2Ffoen%2Fudb066%2F1`
 for versioned.

**Comment by herrstucki** [comment]
@l00mi Let's use a `cube` query param, e.g. `https://visualize.admin.ch/en/create/new?cube=https%3A%2F%2Fenvironment.ld.admin.ch%2Ffoen%2Fudb066`. That way we have the possibility to add more options later

**Comment by herrstucki** [comment]
We need to define if both unversioned and versioned IRIs need to be possible. Some options:

- only use the versioned one (which we use now already as starting point to update to a newer version)
- use different params, e.g. `cube` and `cubeVersioned`, so Visualize can know what to expect
- use an unversioned `cube` param and an optional `version` param (not sure if I like this one, seems more complex to query)

**Comment by l00mi** [comment]
I propose to go step, by step.

Lets define the un-versioned variant first, e.g. as you have proposed above. If there is a need to have a deep link to the versioned later on, we can re-evaluate based on the real need? If you think it's better to touch this only once lets go for the 2nd variant?

**Comment by herrstucki** [comment]
Step by step sounds good to me! This shouldn't be a big deal on our side but maybe we can synchronize it with the changes related to validThrough/expires and #76 so we can complete this in one go.

**Comment by ptbrowne** [comment]
@l00mi On our end, it was much simpler not to introduce the unversioned IRI concept (otherwise we would have to introduce this unversioned iri in all our resolvers), and support only the version iri. This is done here : https://github.com/visualize-admin/visualization-tool/pull/170. This means that the cube creator should use the versioned iri. We also think that it would make sense from the cube creator UI point of view since the links are available from a page that shows a specific version anyway, so linking to a page that uses a specific dataset version makes sense.

**Comment by l00mi** [comment]
Hi @ptbrowne, nice to meet you. Absolutely true from Cube-Creator UI it is not necessary to have a un-versioned link, I would though check back with the customer if they have a use for one potentially? Only because I estimate it to be more time consuming to pick the topic up again in a later time. But from our side this would be solved with a deep-link to the versioned.

The other use cases btw. will be the listing here https://int.lindas.admin.ch/datasets/ and on https://opendata.swiss/de where again the versioned link should work.

**Comment by l00mi** [comment]
@ptbrowne please reopen, or create a new issue.

**Comment by Rdataflow** [comment]
sticky note: 
see upstream https://github.com/visualize-admin/visualization-tool/issues/87#issuecomment-1172164938

**Comment by ptbrowne** [comment]
I think the workExample predicate is missing on the `<https://environment.ld.admin.ch/foen/COVID19VaccPersons_v2>` object.

**Comment by l00mi** [comment]
Interesting point, we use the same link to jump from the cube creator to the cube, even it is not uploaded as Published but as Draft.

I am not sure if we should also allow <https://ld.admin.ch/vocabulary/CreativeWorkStatus/Draft> to be found if there is no version in the link. It has the advantage of being more robust, does anything speak against resolve Draft versions too from a UX point of view?

**Comment by l00mi** [comment]
Yes that would be the proposition. If there is nothing speaking against from another point of view.

**Comment by ptbrowne** [comment]
The workExample should be there even on the unversioned iri right ?

**Comment by ptbrowne** [comment]
I do not see why we could not allow it if coming directly on a versioned / unversioned iri. I opened a PR here : https://github.com/visualize-admin/visualization-tool/pull/594

**Comment by l00mi** [comment]
There seems still to be a miss understanding. What you point to in your query `https://environment.ld.admin.ch/foen/COVID19VaccPersons_v2` is not a cube. It is simply a version-history (compare to https://cube.link/#version-history-of-cubes). It should be used to find the latest cube in this history. (You see it at being of type CreativeWork, and not Cube.)

Then we can have a directly link to a cube (which is normally a version inside the history). 

And finally it is also possible to have cubes, which simply don't have a history attached. (Which should also be able to be shown.)
